src/notation/view/partlistmodel.cpp

```diff
@@ -149,6 +152,44 @@ void PartListModel::selectPart(int partIndex)
     emit dataChanged(index(0), index(rowCount() - 1), { RoleIsSelected });
 }
 
+void PartListModel::resetPart(int partIndex)
+{
+    if (!isExcerptIndexValid(partIndex)) {
+        return;
+    }
+
+    if (!m_excerpts[partIndex]->isEmpty()) {
+        std::string question = mu::trc("notation", "Are you sure you want to reset this part?");
+
+        IInteractive::Button btn = interactive()->question("", question, {
+            IInteractive::Button::Yes, IInteractive::Button::No
+        }).standardButton();
+
+        if (btn != IInteractive::Button::Yes) {
+            return;
+        }
+    }
+
+    doResetPart(partIndex);
+}
+
+void PartListModel::doResetPart(int partIndex)
+{
+    if (!isExcerptIndexValid(partIndex)) {
+        return;
+    }
+
+    bool isCurrentNotation = context()->currentNotation() == m_excerpts[partIndex]->notation();
+
+    m_excerpts[partIndex] = masterNotation()->resetReplaceExcerpt(m_excerpts[partIndex]);
+
+    emit dataChanged(index(partIndex), index(partIndex));
+
+    if (isCurrentNotation) {
+        context()->setCurrentNotation(m_excerpts[partIndex]->notation());
```

Comment:
This code looks quite strange. Why do we need to change the current notation?

Reply:
Because the current excerpt has been deleted and replaced with a new one by `masterNotation()->resetReplaceExcerpt(m_excerpts[partIndex]);`